lib/p2p/packets/types/DisconnectingPacket.ts

```diff
@@ -15,6 +18,46 @@ class DisconnectingPacket extends Packet<DisconnectingPacketBody> {
   public get direction() {
     return PacketDirection.Unilateral;
   }
+
+  public static deserialize(binary: Uint8Array): DisconnectingPacket | undefined {
+    const msg = pb.DisconnectingPacket.deserializeBinary(binary).toObject();
+    return DisconnectingPacket.validate(msg) ? DisconnectingPacket.convert(msg) : undefined;
+  }
+
+  private static validate(msg: pb.DisconnectingPacket.AsObject): boolean {
+    return !!(msg.header
+      && msg.header.id
+      && msg.header.hash
+      && !msg.header.reqid
+      && msg.reason
+    );
+  }
+
+  private static convert = (msg: pb.DisconnectingPacket.AsObject): DisconnectingPacket => {
+    return new DisconnectingPacket({
```

Comment:
What do you think about using the native protobuf objects for the packets? Maybe for a future PR? We could keep these Packet classes as wrapper objects, but have them hold a pb object instead of the header & body JSON objects.

Reply:
Alright that's fine, just a thought.